core/drivers/stm32_tamp.c

```diff
@@ -680,19 +2157,41 @@ static TEE_Result stm32_tamp_probe(const void *fdt, int node,
 
 static const struct stm32_tamp_compat mp13_compat = {
 	.nb_monotonic_counter = 2,
-	.tags = TAMP_HAS_REGISTER_SECCFGR | TAMP_HAS_REGISTER_PRIVCFGR,
+	.tags = TAMP_HAS_REGISTER_SECCFGR | TAMP_HAS_REGISTER_PRIVCFGR |
+		TAMP_HAS_REGISTER_ERCFGR | TAMP_HAS_REGISTER_CR3 |
+		TAMP_HAS_REGISTER_ATCR2 | TAMP_HAS_CR2_SECRET_STATUS |
+		TAMP_SIZE_ATCR1_ATCKSEL_IS_4,
+	.int_tamp = int_tamp_mp13,
+	.int_tamp_size = ARRAY_SIZE(int_tamp_mp13),
+	.ext_tamp = ext_tamp_mp13,
+	.ext_tamp_size = ARRAY_SIZE(ext_tamp_mp13),
+	.pin_map = pin_map_mp13,
+	.pin_map_size = ARRAY_SIZE(pin_map_mp13),
 };
 
 static const struct stm32_tamp_compat mp15_compat = {
 	.nb_monotonic_counter = 1,
 	.tags = 0,
+	.int_tamp = int_tamp_mp15,
+	.int_tamp_size = ARRAY_SIZE(int_tamp_mp15),
+	.ext_tamp = ext_tamp_mp15,
+	.ext_tamp_size = ARRAY_SIZE(ext_tamp_mp15),
+	.pin_map = pin_map_mp15,
+	.pin_map_size = ARRAY_SIZE(pin_map_mp15),
 };
 
 static const struct stm32_tamp_compat mp25_compat = {
 	.nb_monotonic_counter = 2,
-	.tags = TAMP_HAS_REGISTER_SECCFGR |
-		TAMP_HAS_REGISTER_PRIVCFGR |
-		TAMP_HAS_RIF_SUPPORT,
+	.tags = TAMP_HAS_REGISTER_SECCFGR | TAMP_HAS_REGISTER_PRIVCFGR |
+		TAMP_HAS_RIF_SUPPORT | TAMP_HAS_REGISTER_ERCFGR |
+		TAMP_HAS_REGISTER_CR3 |	TAMP_HAS_REGISTER_ATCR2 |
+		TAMP_HAS_CR2_SECRET_STATUS,
+	.int_tamp = int_tamp_mp25,
+	.int_tamp_size = ARRAY_SIZE(int_tamp_mp25),
+	.ext_tamp = ext_tamp_mp25,
+	.ext_tamp_size = ARRAY_SIZE(ext_tamp_mp25),
+	.pin_map = pin_map_mp25,
+	.pin_map_size = ARRAY_SIZE(pin_map_mp25),
 };
```

Comment:
The MP13 and MP25 compat references to `int_tamp_mp*`, `ext_tamp_mp*` and `pin_map_mp*` consume useless internal RAM resident memory when pager is enabled on MP15. Could you add some `CFG_STM32MPxx` directives to prevent that?

Reply:
Thanks